src/app/functions/universal/getFreeScanSearchParams.ts

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ExperimentData } from "../../../telemetry/generated/nimbus/experiments";
+import { modifyAttributionsForUrlSearchParams } from "../universal/attributions";
+
+export type SearchParamArgs = {
+  cookies: {
+    attributionsFirstTouch?: string;
+  };
+  metricsFlowData: MetricFlowData | null;
+  emailInput?: string;
+};
+
+export type MetricFlowParams = {
+  entrypointExperiment: string;
+  entrypointVariation: string;
+  entrypoint: string;
+};
+
+export type MetricFlowData = {
+  deviceId: string;
+  flowId: string;
+  flowBeginTime: number;
+};
+
+export const FREE_SCAN_UTM_SEARCH_PARAMS = {
+  utm_source: "product",
+  utm_medium: "monitor",
+  utm_campaign: "get_free_scan",
+};
+
+export function getFreeScanSearchParams({
+  cookies,
+  emailInput,
+  entrypoint,
+  metricsFlowData,
+  experimentData,
+}: SearchParamArgs &
+  Omit<MetricFlowParams, "entrypointExperiment" | "entrypointVariation"> & {
+    experimentData?: ExperimentData;
+  }) {
+  const attributionSearchParams = modifyAttributionsForUrlSearchParams(
+    new URLSearchParams(cookies.attributionsFirstTouch),
+    {
+      entrypoint,
+      form_type: typeof emailInput === "string" ? "email" : "button",
+      ...(emailInput && { email: emailInput }),
+      ...(metricsFlowData && {
+        device_id: metricsFlowData.deviceId,
+        flow_id: metricsFlowData.flowId,
+        flow_begin_time: metricsFlowData.flowBeginTime.toString(),
+      }),
+      ...(experimentData &&
+        experimentData["landing-page-free-scan-cta"].enabled && {
+          entrypoint_experiment: "landing-page-free-scan-cta",
+          entrypoint_variation:
+            experimentData["landing-page-free-scan-cta"].variant,
+        }),
+    },
+    FREE_SCAN_UTM_SEARCH_PARAMS,
```

Comment:
It might be good to only pass the fallback UTM params when no UTM params are found in `cookies.attributionsFirstTouch`, because as currently written the fallback UTM params can potentially be mixed with UTM params present in `cookies.attributionsFirstTouch` if only some of the source/medium/campaign UTM params are in `cookies.attributionsFirstTouch` (it'll essentially fill in any gaps, potentially leading to unexpected results).